store/mongo/datastore_mongo.go

```diff
@@ -311,14 +309,55 @@ func (db *DataStoreMongo) UpsertDevicesAttributesWithRevision(
 	return db.upsertAttributes(ctx, devices, attrs, false, true, "", "")
 }
 
+func (db *DataStoreMongo) inventoryNeedsUpdate(
+	ctx context.Context,
+	id model.DeviceID,
+	newAttributes model.DeviceAttributes,
+) (rc bool) {
+	rc = false
+	device, err := db.GetDevice(ctx, id)
```

Comment:
Could we get all the devices in one batch?

Reply:
sure, as soon as I am done with what I am tackling now.

Reply:
@alfrunes something like that: https://github.com/mendersoftware/inventory/pull/396/commits/54643f8af17d8073a503ce48a7ff05e85f8f1e13 ? I am fixing some tests now and adding more, byt could you give me a green light on that approach?